### PR TITLE
Prune tautological and duplicate tests

### PR DESCRIPTION
**Pull request by nicholasjconn**
## Summary

Conservative test-suite pruning across the repo. Removed 44 low-value tests (~2.2% of suite, ~1.4% of test LOC) across 22 files; deleted 3 files that became empty or fully redundant.

**Coverage held at 96.66%** (threshold 95%, full `make coverage` run as part of pre-commit).

## What was removed

- **Strict-duplicate tests with weaker assertions** — kept the version that asserts on error message text, removed the one that just asserted `err != nil`.
- **No-assertion tests** — `t.Log` / "doesn't panic" with no `t.Fail*`.
- **Pure tautologies** — re-asserting a constant that the function literally returns (e.g. `previewModelWarningText`).
- **Stdlib-only tests** — exercise only `term.IsTerminal`, `filepath.Join`, etc. with no project logic.
- **Three pointer-trivia tests** consolidated into one table-driven test (`TestIsAgentEnabled`).

## What was deliberately preserved

- All `*_coverage_test.go` files except `internal/clients/codex/launch_coverage_test.go` (which had a single redundant test). Empirical coverage diffs proved the rest exercise unique error-injection branches.
- All migration / rollback / snapshot tests in `internal/install/upgrade_*`.
- All policy, validator, and config-validation tests.
- Sole-coverage tests for any function (verified per-package).

## Files deleted

- `internal/clients/codex/launch_coverage_test.go`
- `internal/install/install_warnings_test.go`
- `internal/projection/mcp_extra_test.go`

## Methodology

10 parallel sub-agents pruned by package with shared conservative criteria (tautology / duplicate / mock-only / compile-check / stdlib-only) and explicit preservation rules (sole-coverage, error injection, edge cases, regression context, table-driven cases). Every removal was followed by a `go test`/`go vet` confirmation, and an independent post-pruning audit (2-round) verified zero orphaned helpers, zero unused imports, zero stale `t.Run` references, and surviving coverage for every affected production function.

## Test plan

- [x] `go test ./...` passes (all 30 packages)
- [x] `go vet ./...` clean
- [x] `goimports -l` empty on all changed files
- [x] `make coverage` reports 96.66% (above 95% threshold)
- [x] Pre-commit hooks (gofmt, lint, tidy, full test suite) pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: internal/root/root_coverage_test.go [hunk 1]
```diff
@@ -7,99 +7,47 @@ import (
...
-		t.Fatalf("FindRepoRoot error: %v", err)
+func TestFindAgentLayerRoot_StatError(t *testing.T) {
+	if os.PathSeparator == '\\' {
+		t.Skip("skipping permission test on windows")
 	}
-	if got != root {
-		t.Fatalf("expected root %s, got %s", root, got)
+	if os.Geteuid() == 0 {
+		t.Skip("skipping permission test when running as root")
 	}
```
[the remark this answers is not in the document]
Comment (nicholasjconn) [reply internal/root/root_coverage_test.go:16]: **No change — incorrect premise.** `os.Geteuid()` is part of the cross-platform `os` package and compiles on Windows (where it returns `-1`). Verified via `GOOS=windows GOARCH=amd64 go build` on a minimal program that calls it — exit 0. Cross-compiling the `internal/root` test binary (`GOOS=windows go test -c ./internal/root/...`) only fails on `syscall.Mkfifo` in the pre-existing `root_test.go:139`; it does not flag `os.Geteuid` in this file. So Geteuid is not the source of any Windows compile issue, and adding a build-tagged helper would be cargo-culting around a non-problem.

File: internal/root/root_coverage_test.go [hunk 2]
```diff
@@ -7,99 +7,47 @@ import (
...
 }
 
 func TestFindRepoRoot_StatError(t *testing.T) {
-	if os.PathSeparator == '\\' { // Corrected escaping for backslash
+	if os.PathSeparator == '\\' {
 		t.Skip("skipping permission test on windows")
 	}
+	if os.Geteuid() == 0 {
+		t.Skip("skipping permission test when running as root")
+	}
```
[the remark this answers is not in the document]
Comment (nicholasjconn) [reply internal/root/root_coverage_test.go:38]: **No change — incorrect premise.** Same as the line-16 reply: `os.Geteuid()` is exported by the standard `os` package on every supported GOOS (returning `-1` on Windows). I verified by cross-compiling a one-line test program (`GOOS=windows go build`) — passes — and by attempting `GOOS=windows go test -c ./internal/root/...`, which only fails on the pre-existing `syscall.Mkfifo` in `root_test.go:139`, never on `os.Geteuid`. No build tag is needed.
